### PR TITLE
Fix entity detaching from head joint not becoming visible

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
### Test Plan

Tests attaching and unattaching an entity to avatar via parenting.

Default wooden mannequin avatar, in front of a mirror, in a domain where you have rez permissions.

Test script files attached.

1) Desktop mode. Test that an entity is not visible to you when attached to your head but becomes visible again when detached from your head.
   1. First person view.
      * Inspect and run testUnparentingFromHeadA.js. Expected result: The entity should be visible in both the Interface window and the mirror, except when it's attached to your avatar's head at which time it should be visible only in the mirror.
      * Inspect and run testUnparentingFromHeadB.js. Expected result: Ditto.
   2. Third person view.
      * Inspect and run testUnparentingFromHeadA.js. Expected result: The entity should be visible in both the Interface window and the mirror at all times.
      * Inspect and run testUnparentingFromHeadB.js. Expected result: Ditto.

2) Repeat test 1) in HMD mode.

[21892.zip](https://github.com/highfidelity/hifi/files/2104817/21892.zip)

https://github.com/highfidelity/hifi/pull/13384

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10604/HighFidelity-Beta-PR13384-249f727.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10604/HighFidelity-Beta-PR13384-249f727.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10604/HighFidelity-Beta-Interface-PR13384-249f727.dmg

**Comment by QliCknox** [comment]
Currently Testing

**Comment by QliCknox** [comment]
Fail @conklin94122 

In Mac Desktop mode, there is consistent crashing while standing in front of a Mirror and running testUnparentingFromHeadA.js and testUnparentingFromHeadB.js in both first and third person mode. This does not occur on Stable when running the same script.
[ScriptCrash.txt](https://github.com/highfidelity/hifi/files/2107067/ScriptCrash.txt)



[Crash.zip](https://github.com/highfidelity/hifi/files/2107082/Crash.zip)

**Comment by ctrlaltdavid** [comment]
@QliCknox Could you please test dev build 8441 on Mac to see if it crashes?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10640/HighFidelity-Beta-PR13384-be4b929.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10640/HighFidelity-Beta-PR13384-be4b929.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10640/HighFidelity-Beta-Interface-PR13384-be4b929.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10640/HighFidelity-Beta-PR13384-be4b929.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10640/HighFidelity-Beta-Interface-PR13384-be4b929.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Mac does Not crash and previous test plan passes.

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -544,16 +548,42 @@ void MyAvatar::simulate(float deltaTime) {
...
+                // Redisplay cauterized children that are not longer children of the head.
+                updateChildCauterization(object, false);
+                object->forEachDescendant([&](SpatiallyNestablePointer descendant) {
+                    updateChildCauterization(descendant, false);
+                });
+                _cauterizedChildrenOfHead.erase(object);
             }
         });
+
         _cauterizationNeedsUpdate = false;
```
[the remark this answers is not in the document]
Comment (ctrlaltdavid) [reply interface/src/avatar/MyAvatar.cpp]: Good point. Done.

**Comment by ctrlaltdavid** [comment]
Retest needed after code change.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10855/HighFidelity-Beta-PR13384-af3861d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10855/HighFidelity-Beta-PR13384-af3861d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10855/HighFidelity-Beta-Interface-PR13384-af3861d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10855/HighFidelity-Beta-PR13384-af3861d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10855/HighFidelity-Beta-Interface-PR13384-af3861d.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Scripts look good.

**Pull request merged**